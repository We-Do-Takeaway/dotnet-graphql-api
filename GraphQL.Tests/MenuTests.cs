using System.Threading.Tasks;
using HotChocolate.Execution;
using Snapshooter.Xunit;
using Xunit;

namespace WeDoTakeawayAPI.GraphQL.Tests
{
    public class MenuTests : BaseTests
    {
        [Fact]
        public async Task Get_Menus()
        {
            var result = await ServiceProvider.ExecuteRequestAsync(
                QueryRequestBuilder
                    .New()
                    .SetQuery(@"
                        query Menus {
                          menus {
                            nodes {
                              id
                              name
                              sections {
                                nodes {
                                  id
                                  name
                                  displayOrder
                                }
                              }
                            }
                         }
                        }")
                    .Create());
        
           result.MatchSnapshot();
            
        }
        
        [Fact]
        public async Task Get_Menu_By_Id()
        {
            var result = await ServiceProvider.ExecuteRequestAsync(
                QueryRequestBuilder
                    .New()
                    .SetQuery(@"
                        query MenuById($id: ID!) {
                          menuById(id: $id) {
                            id
                            name
                          }
                        }")
                    .SetVariableValue(name: "id", value: "600dca30-c6e2-4035-ad15-783c122d6ea1")
                    .Create());
        
            result.MatchSnapshot();
            
        }

        [Fact]
        public async Task Get_Sections_InMenu()
        {
            var result = await ServiceProvider.ExecuteRequestAsync(
                QueryRequestBuilder
                    .New()
                    .SetQuery(@"
                        query MenuWithSections($id: ID!) {
                          menuById(id: $id) {
                            id
                            name
                            sections {
                              nodes {
                                id
                                name
                                displayOrder
                              }
                            }
                          }
                        }")
                    .SetVariableValue(name: "id", value: "600dca30-c6e2-4035-ad15-783c122d6ea1")
                    .Create());
        
            result.MatchSnapshot();
        } 
    }
}