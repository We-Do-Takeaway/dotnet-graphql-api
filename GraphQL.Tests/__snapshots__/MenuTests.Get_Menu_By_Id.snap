{
  "Label": null,
  "Path": null,
  "Data": {
    "menuById": {
      "id": "600dca30-c6e2-4035-ad15-783c122d6ea1",
      "name": "Cafe comfort and waffles"
    }
  },
  "Errors": null,
  "Extensions": null,
  "ContextData": null,
  "HasNext": null
}
